tests/plots/test_index.py

```diff
@@ -267,13 +267,25 @@ def test_generates_index_plot_with_group_filter(self, test_data):
             df,
             value_col="sales",
             group_col="category",
-            index_col="category",
-            value_to_index="A",
+            index_col="region",
+            value_to_index="North",
             include_only_groups=["A", "B"],
         )
 
         assert isinstance(result_ax, plt.Axes)
 
+        # Verify that only the filtered groups appear in the plot
+        y_labels = [label.get_text() for label in result_ax.get_yticklabels()]
+        plotted_groups = set(y_labels)
+        expected_groups = {"A", "B"}
+
+        assert plotted_groups.issubset(expected_groups), (
```

Comment:
Can we strength this assertion a bit.
```suggestion
        assert plotted_groups == expected_groups, (
```